web/packages/teleport/src/JoinTokens/CreateJoinTokenDialog.tsx

```diff
@@ -0,0 +1,565 @@
+                isMulti
+                isSimpleValue
+                mb={5}
+                isClearable={false}
+                value={newTokenState.roles}
+                onChange={setTokenRoles}
+                options={joinRoleOptions}
+              />
+              {newTokenState.roles.some(i => i.value === 'Bot') && ( // if Bot is included, we must get a bot name as well
+                <FieldInput
+                  label="Bot name"
+                  toolTipContent="Bot names are required when the Bot role is selected"
+                  rule={requiredField('Bot name is required')}
+                  placeholder="Enter bot name"
+                  value={newTokenState.bot_name}
+                  onChange={e => setTokenField('bot_name', e.target.value)}
+                />
+              )}
+              {newTokenState.method.value === 'iam' && (
+                <JoinTokenIAMForm
+                  tokenState={newTokenState}
+                  onUpdateState={newState => setNewTokenState(newState)}
+                />
+              )}
+              {newTokenState.method.value === 'gcp' && (
+                <JoinTokenGCPForm
+                  tokenState={newTokenState}
+                  onUpdateState={newState => setNewTokenState(newState)}
+                />
+              )}
+              <Flex
+                mt={4}
+                py={4}
+                gap={2}
+                css={`
+                  position: sticky;
+                  bottom: 0;
+                  background: ${({ theme }) => theme.colors.levels.sunken};
+                  border-top: 1px solid
+                    ${props => props.theme.colors.spotBackground[1]};
+                `}
+              >
+                <ButtonPrimary
+                  width="100%"
+                  size="large"
+                  textTransform="none"
+                  onClick={() => save(validator)}
+                  disabled={createTokenAttempt.status === 'processing'}
+                >
+                  {editToken ? 'Edit' : 'Create'} Join Token
+                </ButtonPrimary>
+                <ButtonSecondary
+                  width="100%"
+                  textTransform="none"
+                  size="large"
+                  onClick={() => {
+                    reset(validator);
+                    onClose();
+                  }}
+                  disabled={false}
+                >
+                  Cancel
+                </ButtonSecondary>
+              </Flex>
+            </Box>
+          )}
+        </Validation>
+      </Box>
+    </Flex>
+  );
+};
+
+const RuleBox = styled(Box)`
+  border-color: ${props =>
+    props.theme.colors.interactive.tonal.neutral[0].background};
+  border-width: 2px;
+  border-style: solid;
+  border-radius: ${props => props.theme.radii[2]}px;
+
+  margin-bottom: ${props => props.theme.space[3]}px;
+
+  padding: ${props => props.theme.space[3]}px;
+`;
+
+const JoinTokenIAMForm = ({
```

Comment:
nit: i would pull these forms (JoinTokenIAMForm and  JoinTokenGCPForm) out into its own file?